src/webhook/test_regular_user_validation.py

```diff
@@ -40,34 +35,34 @@ def __getitem__(self, item):
 
 class TestRegularUserValidation(unittest.TestCase):
     def test_is_request_allowed_unauthenticated(self):
-        self.assertFalse(is_request_allowed("system:unauthenticated", "", ADMIN_GROUPS))
+        self.assertFalse(is_request_allowed("system:unauthenticated", [], ADMIN_GROUPS))
 
     def test_is_request_allowed_sre(self):
-        self.assertTrue(is_request_allowed("nmalik@redhat.com", ADMIN_GROUPS[0], ADMIN_GROUPS))
+        self.assertTrue(is_request_allowed("nmalik@redhat.com", [ADMIN_GROUPS[0]], ADMIN_GROUPS))
 
     def test_is_request_allowed_kubeadmin(self):
-        self.assertTrue(is_request_allowed("kube:admin", "", ADMIN_GROUPS))
+        self.assertTrue(is_request_allowed("kube:admin", [], ADMIN_GROUPS))
 
     def test_is_request_allowed_systemadmin(self):
-        self.assertTrue(is_request_allowed("system:admin", "", ADMIN_GROUPS))
+        self.assertTrue(is_request_allowed("system:admin", [], ADMIN_GROUPS))
 
     def test_is_request_allowed_nonadmin_group(self):
-        self.assertFalse(is_request_allowed("someuser", "some-group", ADMIN_GROUPS))
+        self.assertFalse(is_request_allowed("someuser", ["some-group"], ADMIN_GROUPS))
```

Comment:
Missing tests with variations on `admin_groupnames`, including permutations where `groupnames` contains groups that would normally be allowed under `ADMIN_GROUPS`. Also permutations proving that `system:*`/`kube:*` is allowed because it's hitting that code path rather than the one that passes based on groups. (I should have caught that in PR #40, sorry.)

Reply:
Latest commit should cover more permutations, thanks.